app/frontend/components/pages/accounts/accounts.tsx

```diff
@@ -0,0 +1,286 @@
+  showSendTransactionModal,
+  showDelegationModal,
+  shouldShowSendTransactionModal,
+  shouldShowDelegationModal,
+  selectedAccount,
+  totalWalletBalance,
+  totalRewardsBalance,
+}) => {
+  const accountInfos = Object.values(accounts)
+  const firstAddressPerAccount = accountInfos.map((e: any) => e.visibleAddresses[0].address)
+  const InfoAlert = () => (
+    <Fragment>
+      <div className="dashboard-column account sidebar-item info">
+        <Alert alertType="info sidebar">
+          <p>
+            <strong>Accounts</strong> offer a way to split your funds. You are able to delegate to
+            different pool from each account. Each account has different set of addresses and keys.
+          </p>
+        </Alert>
+      </div>
+      <div className="dashboard-column account sidebar-item info">
+        <Alert alertType="info sidebar">
+          <p>
+            Click explore/activate button to load data for related account. If you are using
+            hardware wallet, you will be requested to export public key.
+          </p>
+        </Alert>
+      </div>
+      <div className="dashboard-column account info">
+        <Alert alertType="warning sidebar">
+          <p>
+            This feature might not be supported on other wallets yet. If you decide to move your
+            funds to <strong>account</strong> other then <strong>account 0</strong>, you might not
+            see your funds in other wallets.
+          </p>
+        </Alert>
+      </div>
+    </Fragment>
+  )
+
+  return (
+    <Fragment>
+      {shouldShowSendTransactionModal && <SendTransactionModal />}
+      {shouldShowDelegationModal && <DelegationModal />}
+      <div className="dashboard-column account">
+        <div className="card account-aggregated">
+          <div className="balance">
+            <div className="item">
+              <h2 className="card-title small-margin">Total balance</h2>
+              <div className="balance-amount">
+                {printAda(totalWalletBalance as Lovelace)}
+                <AdaIcon />
+              </div>
+            </div>
+            <div className="item">
+              <h2 className="card-title small-margin">Total rewards balance</h2>
+              <div className="balance-amount">
+                {printAda(totalRewardsBalance as Lovelace)}
+                <AdaIcon />
+              </div>
+            </div>
+          </div>
+          <div className="refresh-wrapper">
+            <button className="button secondary balance refresh" onClick={reloadWalletInfo}>
+              Refresh
+            </button>
+          </div>
+        </div>
+        <div className="mobile">
+          <InfoAlert />
+        </div>
+        <div className="accounts-wrapper">
+          <div className="dashboard-column account list">
+            <div>
+              {range(0, Object.keys(accounts).length + 1).map(
+                (i) =>
+                  (i === 0 || accounts[i - 1].isUsed) && (
```

Comment:
shouldn't we just break the loop as soon as we come across the first unused account? Otherwise we wouldn't be wouldn't be BIP44 compliant: https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#account-discovery in the sense that we shouldn't let the users use two non-consecutive accounts if no account in between is used as the discovery is supposed to stop as soon as the currently explored account is unused

Reply:
We are BIPP 44 compliant, the current behavior is the exact as you explained. Let's say a wallet has 3 used accounts, in that case, we want to display 4 "AccountTiles" the last one to be explored if needed. If a user explores this account we would still display 4 account tiles until the user doesn't use the 4th account. But I agree this doesn't look very good. I refactored so we calculate the number of user accounts in advance.

```
{range(0, usedAccountsCount + 1).map(
                (i) =>
                  (

Reply:
hmm, `usedAccountsCount` seems to be used only here and if we are really BIP44 compliant, it should always be one less than the total number of accounts shown to the user. Wouldn't it make sense to just iterate over all the accounts loaded into the state?

Also, I'm considering whether it's actually a good idea to have as an `accounts` object and with its keys being their hardened indices. It seems more natural to me to just have an array of accounts and not rely on `Object.values` and the index can just be it's unhardened value (0, 1 etc) which in turn is more readable when debugging, etc

Reply:
You removed the part regarding the accounts being an object and not an array, I was planning to refactor it but didnt get to it today. 

Regarding the usedAccountCount. `it should always be one less than the total number of accounts shown to the user.`, well that's exactly how it is now. There is a difference between the number of accounts in the state and the number of accounts used. We can call about this tomorrow.